cli/src/commands/parse.rs

```diff
@@ -135,11 +135,13 @@ pub fn parse_lockfile(
     }
 
     // Abort if generation is disabled or path is neither lockfile nor manifest.
-    if !generate_lockfiles || !parser.is_path_manifest(&path) {
+    let maybe_manifest = parser.is_path_manifest(&path);
+    if !generate_lockfiles || !maybe_manifest {
```

Comment:
nit/imo: I think this is slightly easier to read if you apply DML.
```suggestion
    if !(generate_lockfiles && maybe_manifest) {
```